[Elixir] Add analyzer comments for boutique-inventory (#2101)
exercism/website-copy: analyzer-comments/elixir/boutique-inventory/use_enum_reduce.md

# Use `Enum.reduce`

> One purpose of this exercise it to learn about reduce. Use the `Enum.reduce` function in the `total_quantity` function to practice.

Reviewer: ```suggestion
Use the `Enum.reduce` function in the `total_quantity` function to practice. It's the best fitting `Enum` function for this task.
```
